### PR TITLE
Travis-native Headless Chrome for WDIO

### PR DESCRIPTION
**Pull request by AronNovak**
#160 

## Status

ported https://github.com/Gizra/drupal-elm-starter/pull/162

### CODE REVIEW

**Comment by AronNovak** [comment]
**0.5/3h**

The integration itself should work, what's left:
 - DB upload
 - double-check failure detection

**Comment by AronNovak** [comment]
Status from Travis:
image

File: ci-scripts/prepare_client.sh [hunk 1]
```diff
@@ -1,16 +1,27 @@
...
-if [ -z "${BUILD_CLIENT+x}" ] || [ "$BUILD_CLIENT" -ne 1 ]; then
+if [ -z "${BUILD_WEBDRIVERIO+x}" ] || [ "$BUILD_WEBDRIVERIO" -ne 1 ]; then
  exit 0;
 fi
 
-"$TRAVIS_BUILD_DIR"/sysconfcpus/bin/sysconfcpus -n 2 elm-make --yes
-cp ./client/src/elm/LocalConfig.Example.elm ./client/src/elm/LocalConfig.elm
+# Install global packages.
+npm install -g elm@~0.18.0
+npm install -g elm-test
```
Comment (amitaibu) [inline ci-scripts/prepare_client.sh]: I think we should lock this version

**Comment by AronNovak** [comment]
image

It seems headless Chrome finally operates correctly to execute our WDIO tests, fixing the rest:
 - restore / improve caching
 - test robot with DB dump upon failure

File: ci-scripts/prepare_client.sh (hunk 1 above)
Comment (AronNovak) [reply ci-scripts/prepare_client.sh]: Absolutely, and even more, `elm-test` is not needed for executing WDIO tests, just removed

**Comment by AronNovak** [comment]
current PR:
image

master branch:
image

There are couple of small issues to fix, but the results are promising, the overhead is lot less this way.

**1.5/3h**

**Comment by AronNovak** [comment]
```shell
#!/usr/bin/env bash
set -e

set +o errexit
set -o pipefail
node_modules/.bin/wdio wdio.conf.js | tee /tmp/moo.txt
WDIO_CMD_RET=$?
if [[ "$WDIO_CMD_RET" -eq 0 ]]; then
  echo "pass"
  exit
fi
echo "failed"
˛```

With a tiny local script, could reproduce and fix the error detection issue.

**Comment by AronNovak** [comment]
**1.75/3h**

**Comment by AronNovak** [comment]
image

**2/3h**

**Comment by AronNovak** [comment]
Oh my friend, @gizra-robot , finally you arrived to this PR! :)

**Comment by AronNovak** [comment]
image

All green, quite okay wall clock times still.
**2.5/3h**

**Comment by AronNovak** [comment]
Merging it now to the base PR, continuing the shell script structuring there.

**Pull request merged**